### DISCUSSION TITLE
SOS

### DISCUSSION THREAD

**Discussion by QiaoTuCodes** [micropython/micropython#9768] [Using MicroPython] [answered]
Dear creator 

In minimal ports . i'll try compile firmware to stm32f410rb series mcu core. the douctment mentioned about how to disable MICROPY_ENABLE_COMPILER. it did it! but got a new problem: how can i  mpy files? i just try run : mpy-cross xxx.py make mpy files was success!. but if try : make CROSS=1 was unsuccess in the final say:

`MISC freezing bytecode
CC build/_frozen_mpy.c
build/_frozen_mpy.c:50:28: error: expected '=', ',', ';', 'asm' or '__attribute__' before '.' token
 STATIC const byte fun_data_.._.._micropython_ports_STM32F400RBT6_frozentest__lt_module_gt_[85] = {
                            ^
build/_frozen_mpy.c:93:37: error: expected '=', ',', ';', 'asm' or '__attribute__' before '.' token
 STATIC const mp_obj_str_t const_obj_.._.._micropython_ports_STM32F400RBT6_frozentest__lt_module_gt__0 = {{&mp_type_str}, 246, 34, (const byte*)"\x61\x20\x6c\x6f\x6e\x67\x20\x73\x74\x72\x69\x6e\x67\x20\x74\x68\x61\x74\x20\x69\x73\x20\x6e\x6f\x74\x20\x69\x6e\x74\x65\x72\x6e\x65\x64"};
                                     ^
build/_frozen_mpy.c:94:37: error: expected '=', ',', ';', 'asm' or '__attribute__' before '.' token
 STATIC const mp_obj_str_t const_obj_.._.._micropython_ports_STM32F400RBT6_frozentest__lt_module_gt__1 = {{&mp_type_str}, 200, 38, (const byte*)"\x61\x20\x73\x74\x72\x69\x6e\x67\x20\x74\x68\x61\x74\x20\x68\x61\x73\x20\x75\x6e\x69\x63\x6f\x64\x65\x20\xce\xb1\xce\xb2\xce\xb3\x20\x63\x68\x61\x72\x73"};
                                     ^
build/_frozen_mpy.c:95:37: error: expected '=', ',', ';', 'asm' or '__attribute__' before '.' token
 STATIC const mp_obj_str_t const_obj_.._.._micropython_ports_STM32F400RBT6_frozentest__lt_module_gt__2 = {{&mp_type_bytes}, 57, 11, (const byte*)"\x62\x79\x74\x65\x73\x20\x31\x32\x33\x34\x01"};
                                     ^
build/_frozen_mpy.c:96:44: error: expected '=', ',', ';', 'asm' or '__attribute__' before '.' token
 STATIC const mp_rom_obj_t const_table_data_.._.._micropython_ports_STM32F400RBT6_frozentest__lt_module_gt_[3] = {
                                            ^
build/_frozen_mpy.c:101:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '.' token
 const mp_raw_code_t raw_code_.._.._micropython_ports_STM32F400RBT6_frozentest__lt_module_gt_ = {
                              ^
build/_frozen_mpy.c:141:6: error: 'raw_code_' undeclared here (not in a function); did you mean 'mp_raw_code_t'?
     &raw_code_.._.._micropython_ports_STM32F400RBT6_frozentest__lt_module_gt_,
      ^~~~~~~~~
      mp_raw_code_t
build/_frozen_mpy.c:141:16: error: expected identifier before '.' token
     &raw_code_.._.._micropython_ports_STM32F400RBT6_frozentest__lt_module_gt_,
                ^
../../py/mkrules.mk:77: recipe for target 'build/build/_frozen_mpy.o' failed
make: *** [build/build/_frozen_mpy.o] Error 1
`
what happend that?

**Comment by jimmo** [comment, marked as answer]
I think what's happened here is you've used a relative path (i.e. `../../path/to/xxx.py`) and this has ended up in the mpy file. mpy-tool should probably handle this, but in the meantime you can use the `-s` argument to mpy-cross to override the source name. e.g. `mpy-cross -o xxx.mpy -s xxx.py ../../path/to/xxx.py`.

**Reply by QiaoTuCodes** [reply]
Thank you Jim. my probem was done. just Absolute path & relative path that. Thank you again.

**Comment by QiaoTuCodes** [comment]
okey .thank you Jim. my probem was done. just&nbsp;Absolute path &amp; relative path that. Thank you again.
&nbsp;


深藍許願樹
***@***.***



&nbsp;
